vsphere/resource_vsphere_tag_category.go

```diff
@@ -0,0 +1,251 @@
+	vSphereTagCategoryCardinalityMultiple = "MULTIPLE"
+
+	vSphereTagCategoryAssociableTypeFolder                         = "Folder"
+	vSphereTagCategoryAssociableTypeClusterComputeResource         = "ClusterComputeResource"
+	vSphereTagCategoryAssociableTypeDatacenter                     = "Datacenter"
+	vSphereTagCategoryAssociableTypeDatastore                      = "Datastore"
+	vSphereTagCategoryAssociableTypeStoragePod                     = "StoragePod"
+	vSphereTagCategoryAssociableTypeDistributedVirtualPortgroup    = "DistributedVirtualPortgroup"
+	vSphereTagCategoryAssociableTypeDistributedVirtualSwitch       = "DistributedVirtualSwitch"
+	vSphereTagCategoryAssociableTypeVmwareDistributedVirtualSwitch = "VmwareDistributedVirtualSwitch"
+	vSphereTagCategoryAssociableTypeHostSystem                     = "HostSystem"
+	vSphereTagCategoryAssociableTypeContentLibrary                 = "com.vmware.content.Library"
+	vSphereTagCategoryAssociableTypeContentLibraryItem             = "com.vmware.content.library.Item"
+	vSphereTagCategoryAssociableTypeHostNetwork                    = "HostNetwork"
+	vSphereTagCategoryAssociableTypeNetwork                        = "Network"
+	vSphereTagCategoryAssociableTypeOpaqueNetwork                  = "OpaqueNetwork"
+	vSphereTagCategoryAssociableTypeResourcePool                   = "ResourcePool"
+	vSphereTagCategoryAssociableTypeVirtualApp                     = "VirtualApp"
+	vSphereTagCategoryAssociableTypeVirtualMachine                 = "VirtualMachine"
+
+	vSphereTagCategoryAssociableTypeAll = "All"
+)
+
+// The following groups are type groups that are associated with the same type
+// selection in the vSphere Client tag category UI.
+var (
+	// vSphereTagCategoryAssociableTypesForDistributedVirtualSwitch represents
+	// types for virtual switches.
+	vSphereTagCategoryAssociableTypesForDistributedVirtualSwitch = []string{
+		vSphereTagCategoryAssociableTypeDistributedVirtualSwitch,
+		vSphereTagCategoryAssociableTypeVmwareDistributedVirtualSwitch,
+	}
+
+	// vSphereTagCategoryAssociableTypesForNetwork represents the types for
+	// networks.
+	vSphereTagCategoryAssociableTypesForNetwork = []string{
+		vSphereTagCategoryAssociableTypeHostNetwork,
+		vSphereTagCategoryAssociableTypeNetwork,
+		vSphereTagCategoryAssociableTypeOpaqueNetwork,
+	}
+)
+
+func resourceVSphereTagCategory() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceVSphereTagCategoryCreate,
+		Read:   resourceVSphereTagCategoryRead,
+		Update: resourceVSphereTagCategoryUpdate,
+		Delete: resourceVSphereTagCategoryDelete,
+		Importer: &schema.ResourceImporter{
+			State: resourceVSphereTagCategoryImport,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:        schema.TypeString,
+				Description: "The display name of the category.",
+				Required:    true,
+			},
+			"description": {
+				Type:        schema.TypeString,
+				Description: "The description of the category.",
+				Optional:    true,
```

Comment:
should this also be Computed?

Reply:
Config should be the source of truth, even if there's no `description` defined, so this should not be computed so that it can be reverted to a blank description if set out-of-band.